internal/services/riotclient.go

```diff
@@ -0,0 +1,57 @@
+package services
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"riot-developer-proxy/internal/valueobjects"
+)
+
+type RiotClient struct {
+	client    http.Client
+	baseURI   string
+	riotToken *string
+}
+
+func NewRiotClient(client http.Client, baseURI string) *RiotClient {
+	return &RiotClient{
+		client:  client,
+		baseURI: baseURI,
+	}
+}
+
+func (rc *RiotClient) WithToken(token string) {
+	rc.riotToken = &token
+}
+
+func (rc *RiotClient) DoReq(ctx context.Context, method string, path string) (*valueobjects.ProxyResponse, error) {
+	req, err := http.NewRequestWithContext(ctx, method, rc.baseURI+path, nil)
+
```

Comment:
nit: usually we don't leave a newline between the function call that returns `err` and the `if err` itself. Same for similar places below

Reply:
Ok Thanks, i'll fix